src/libraries/System.Net.Security/src/System/Net/Security/Pal.OSX/SafeDeleteSslContext.cs

```diff
@@ -144,8 +152,16 @@ protected override void Dispose(bool disposing)
                 SafeSslHandle sslContext = _sslContext;
                 if (null != sslContext)
                 {
-                    _inputBuffer.Dispose();
-                    _outputBuffer.Dispose();
+                    if (Interlocked.Exchange(ref _pendingInput, Disposed) == NoIOPending)
+                    {
+                        _inputBuffer.Dispose();
+                    }
+
+                    if (Interlocked.Exchange(ref _pendingOutput, Disposed) == NoIOPending)
+                    {
+                        _outputBuffer.Dispose();
+                    }
+
                     sslContext.Dispose();
```

Comment:
I realize this PR is focused on on the buffers, but what does it mean to dispose of the sslContext while that's still in use as well?  Is all inappropriate use there handled by this being a SafeHandle, such that actual release of the underlying handle is delayed until all interop with the handle completes (typical SafeHandle stuff)?  If yes, would it make sense to guard that as well?

Reply:
I can take a look. There was previous discussion about using SafeHandles in SslStream. Perhaps that can go together. The buffer part is regression caused by #32338.